lib/ansible/plugins/connection/ssh.py

```diff
@@ -1477,3 +1483,37 @@ def reset(self) -> None:
 
     def close(self) -> None:
         self._connected = False
+
+    @property
+    def has_tty(self):
+        return self.is_tty_requested()
+
+    def _is_tty_requested(self):
+
+        # check if we require tty (only from our args, cannot see options in configuration files)
+        opts = []
+        for opt in ('ssh_args', 'ssh_common_args', 'ssh_extra_args'):
+            attr = self.get_option(opt)
+            if attr is not None:
+                opts.extend(self._split_ssh_args(attr))
+
+        args, dummy = self._tty_parser.parse_known_args(opts)
+
+        if args.t:
+            return True
+
+        for arg in args.o or []:
+            val = arg.split('=', 1)
+            if val[0].lower() == 'requesttty':
+                if val[1].lower() in ('yes', 'force'):
```

Comment:
This parses `"-o 'RequestTTY=force'"` but not `"-o 'RequestTTY force'"`. There is also optional whitespace around the =.

```suggestion
            if '=' in arg:
                val = arg.split('=', 1)
            else:
                val = arg.split(maxsplit=1)

            if val[0].lower().strip() == 'requesttty':
                if val[1].lower().strip() in ('yes', 'force'):
```

Reply:
Should this include 'auto'?

Reply:
no, since auto will always be 'no' for us since we should not be asking for full login but running a command